client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java

```diff
@@ -112,7 +112,11 @@ public List<ShuffleBlockInfo> addRecord(int partitionId, Object key, Object valu
     final long start = System.currentTimeMillis();
     arrayOutputStream.reset();
     serializeStream.writeKey(key, ClassTag$.MODULE$.apply(key.getClass()));
-    serializeStream.writeValue(value, ClassTag$.MODULE$.apply(value.getClass()));
+    if (value != null) {
+      serializeStream.writeValue(value, ClassTag$.MODULE$.apply(value.getClass()));
+    } else {
+      serializeStream.writeValue(null, null);
```

Comment:
Is it consistent with spark logic?

Reply:
> Is it consistent with spark logic?

There is no special handling of null values in spark, but when the value is null, scala infers that the ClassTag is `scala.reflect.ManifestFactory#Null`

![image](https://user-images.githubusercontent.com/17894939/199685686-b3e5bbc2-21d2-433d-bf06-ca8ef7d85045.png)